### ISSUE TITLE
Undeclared dependency Env

### ISSUE THREAD

**Issue by eserte** [plicease/File-Which#13]
The test suite fails if Env.pm is not installed. This may happen on some systems where the system perl is missing standard modules (e.g. on CentOS or RedHat):
```
Can't locate Env.pm in @INC (@INC contains: ... .) at t/02_all.t line 3.
BEGIN failed--compilation aborted at t/02_all.t line 3.
t/02_all.t ...... 
Dubious, test returned 2 (wstat 512, 0x200)
No subtests run 
Can't locate Env.pm in @INC (@INC contains: ... .) at t/03_simple.t line 6.
BEGIN failed--compilation aborted at t/03_simple.t line 6.
# Looks like your test exited with 2 before it could output anything.
t/03_simple.t ... 
Dubious, test returned 2 (wstat 512, 0x200)
Failed 10/10 subtests 
```

**Comment by calid** [comment]
This is really a bug with CentOS, which should install Env as part of the base perl install.  I've reported the issue here:  https://bugs.centos.org/view.php?id=15893

**Comment by calid** [comment]
Think this should be safe to close, per the RHEL package maintainers on Centos/RHEL 7 it is necessary to install perl-core, not perl, if you want perl + core modules.  This will be fixed in Centos/RHEL 8, where the perl package itself will install core as expected:  https://bugzilla.redhat.com/show_bug.cgi?id=1684902

**Comment by eserte** [comment]
Still it probably does not hurt to add "Env" to the prereq list.

**Comment by calid** [comment]
What do you think if instead the module fails during configure and if on centos/rhel 7 prints a message saying something like q(Could not find Env.pm, have you installed perl-core?).  Hardcoding a prereq on a core module silently hides the fact the user's Perl installation is incomplete/broken.  Dying and printing a helpful error informs the user of this and ensures they fix it the right way, by installing perl-core.

**Comment by eserte** [comment]
I think the term "broken" is too harsh here. Having a small perl installation may have its virtues, e.g. for really small systems.

**Comment by Chaz6** [comment]
+1 to register the dependency from me. I like minimal distributions.

**Comment by pplu** [comment]
Stumped on this also: having Env as a dependency would of helped things go inadverted

**Comment by ufobat** [comment]
I stumbled across this issue and it took me way to long to figure out what's going on.

I do agree with you that this is a bug in centos not in your module.

BUT if it is not a bug to declare Env as a dependency then please keep in mind that you could easily document why you have declared it as dependency *and* as a benefit to that you make installing your module easy to beginners *and* improve the "whole perl experience" for other users.

**Comment by plicease** [comment]
Addressed in `1.26_01`
